shell/platform/darwin/ios/framework/Source/FlutterAppDelegateTest.mm

```diff
@@ -78,15 +72,69 @@ - (void)skip_testLaunchUrlWithFragmentNoQueryParameter {
     return viewController;
   };
   NSURL* url = [NSURL URLWithString:@"http://myApp/custom/route#fragment"];
-  NSDictionary<UIApplicationOpenURLOptionsKey, id>* options = @{};
-  BOOL result = [appDelegate application:[UIApplication sharedApplication]
-                                 openURL:url
-                                 options:options
-                         infoPlistGetter:^NSDictionary*() {
-                           return @{@"FlutterDeepLinkingEnabled" : @(YES)};
-                         }];
+  BOOL result = [appDelegate openURL:url
+                     infoPlistGetter:^NSDictionary*() {
+                       return @{@"FlutterDeepLinkingEnabled" : @YES};
+                     }];
   XCTAssertTrue(result);
   OCMVerify([navigationChannel invokeMethod:@"pushRoute" arguments:@"/custom/route#fragment"]);
 }
 
+#pragma mark - Deep linking
+
+- (void)testUniversalLinkWebBrowserUrl {
+  FlutterAppDelegate* appDelegate = [[FlutterAppDelegate alloc] init];
+  FlutterViewController* viewController = OCMClassMock([FlutterViewController class]);
+  FlutterEngine* engine = OCMClassMock([FlutterEngine class]);
+  FlutterMethodChannel* navigationChannel = OCMClassMock([FlutterMethodChannel class]);
+  OCMStub([engine navigationChannel]).andReturn(navigationChannel);
+  OCMStub([viewController engine]).andReturn(engine);
+  // Set blockArg to a strong local to retain to end of scope.
+  id blockArg = [OCMArg invokeBlockWithArgs:@NO, nil];
+  OCMStub([engine waitForFirstFrame:3.0 callback:blockArg]);
+  appDelegate.rootFlutterViewControllerGetter = ^{
+    return viewController;
+  };
+  NSUserActivity* userActivity =
+      [[NSUserActivity alloc] initWithActivityType:NSUserActivityTypeBrowsingWeb];
+  userActivity.webpageURL = [NSURL URLWithString:@"http://myApp/custom/route?query=test"];
+  BOOL result = [appDelegate
+               application:[UIApplication sharedApplication]
+      continueUserActivity:userActivity
+        restorationHandler:^(NSArray<id<UIUserActivityRestoring>>* __nullable restorableObjects){
+        }];
+  XCTAssertFalse(result);
+  OCMReject([navigationChannel invokeMethod:OCMOCK_ANY arguments:OCMOCK_ANY]);
+}
+
+- (void)testUniversalLinkPushRoute {
```

Comment:
These 2 tests are very similar, maybe it's worth considering pulling out a helper function to remove some of the boilerplate so the differences are more apparent.

Reply:
I pulled the boilderplate in all these tests up into `setUp`.  Hopefully it's easier to read now.